src/subscriber.rs

```diff
@@ -155,65 +165,73 @@ async fn handle_paid_invoice(
 
             tracing::info!(
                 event_id = event_id.to_bech32().expect("bech32"),
-                "Broadcasted zap receipt",
+                "Broadcasted zap receipt for game zap",
             );
 
-            tokio::spawn({
-                // broadcast zap receipt to client relays.
-                async move {
-                    if let Err(e) = broadcast_to_client_relays(event, client).await {
-                        tracing::warn!(
-                            "Failed to broadcast zap receipt to client relays! Error: {e:#}"
-                        );
-                    }
-                }
-            });
-
+            Ok(())
+        }
+        Some(_) => {
+            tracing::warn!("Ignoring other zaps which might have been donations.");
             Ok(())
         }
     }
 }
 
-pub async fn broadcast_to_client_relays(zap_receipt: Event, client: Client) -> anyhow::Result<()> {
-    tracing::info!("Broadcasting zap receipt to client relays");
+async fn publish_zap_receipt(keys: &Keys, zap: &mut Zap, client: Client) -> Result<EventId> {
+    let preimage = zap.request.id.to_bytes();
+    let invoice_hash = bitcoin::hashes::sha256::Hash::hash(&preimage);
+
+    let payment_secret = zap.request.id.to_bytes();
+
+    let private_key =
+        SecretKey::from_hashed_data::<bitcoin::hashes::sha256::Hash>(zap.request.id.as_bytes());
+
+    let amt_msats = zap
+        .invoice
+        .amount_milli_satoshis()
+        .expect("Invoice must have an amount");
+
+    let fake_invoice = InvoiceBuilder::new(Currency::Bitcoin)
+        .amount_milli_satoshis(amt_msats)
+        .invoice_description(zap.invoice.description())
+        .current_timestamp()
+        .payment_hash(invoice_hash)
+        .payment_secret(PaymentSecret(payment_secret))
+        .min_final_cltv_expiry_delta(144)
+        .basic_mpp()
+        .build_signed(|hash| {
+            Secp256k1::signing_only().sign_ecdsa_recoverable(hash, &private_key)
+        })?;
+
+    let event = EventBuilder::zap_receipt(
+        fake_invoice.to_string(),
+        Some(hex::encode(preimage)),
+        &zap.request.clone(),
+    )
+    .to_event(keys)?;
+
+    let event_id = client.send_event(event.clone()).await?;
+    Ok(event_id)
+}
 
-    client.send_event(zap_receipt).await?;
-    Ok(())
+async fn ephermal_client(client: Client, zap: &mut Zap) -> anyhow::Result<Client> {
+    let og_client = client.clone();
+    let options = Options::default();
+    let client = Client::with_opts(
+        og_client.signer().await?,
+        options
+            .wait_for_send(true)
+            .send_timeout(Some(Duration::from_secs(20))),
+    );
+    let relays = og_client.relays().await;
+    let relays = relays.keys();
+    client.add_relays(relays).await?;
+    client.add_relays(utils::get_relays(&zap.request)?).await?;
+    client.connect().await;
+    client.set_zapper(og_client.zapper().await?).await;
```

Comment:
🤔 you shouldn't need a zapper if we just want to publish a zap receipt.

Reply:
The client is also used to zap the winners on the other branch